### PR TITLE
Us 45 calendar event invitations

### PR DESCRIPTION
**Pull request by drauf**

### CODE REVIEW

File: src/main/java/TdpRecruitmentApplication.java [hunk 1]
```diff
@@ -45,7 +46,8 @@ public DataSourceFactory getDataSourceFactory(TdpRecruitmentApplicationConfigura
 
     @Override
     public void initialize(Bootstrap<TdpRecruitmentApplicationConfiguration> bootstrap) {
-        bootstrap.addBundle(new AssetsBundle("/assets/", "/", "index.html"));
+        bootstrap.addBundle(new FileAssetsBundle("src/main/resources/assets", "/", "index.html")); //don't merge with master
+//        bootstrap.addBundle(new AssetsBundle("/assets/", "/", "index.html"));
```
Comment (drauf) [inline src/main/java/TdpRecruitmentApplication.java]: revert this before merge

File: src/main/java/services/Interview.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        calendarPart.setContent(createCalendarEvent(), "text/calendar;method=REQUEST");
+        calendarPart.addHeader("Content-Class", "urn:content-classes:calendarmessage");
+        body.addBodyPart(calendarPart);
+
+        return body;
+    }
+
+    private String createCalendarEvent() {
+        String template = "";
+        UUID eventUUID = UUID.randomUUID(); //should this be random?
```
Comment (danielr-kainos) [inline src/main/java/services/Interview.java]: .

File: src/main/java/services/Interview.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        iCalDate.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        URL url = Resources.getResource("email/event_template.ics");
+        try {
+            template = Resources.toString(url, Charsets.UTF_8);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return template.replace("{{organizer}}", parseOrganizer())  //TODO replace room field
```
Comment (danielr-kainos) [inline src/main/java/services/Interview.java]: .

File: src/main/java/services/MailingTask.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        String from = config.getFrom();
+        String password = config.getPassword();
+
+        Properties props = new Properties();
+        props.put("mail.smtp.auth", (password != null) ? "true" : "false");
+        props.put("mail.smtp.starttls.enable", "true");
+        props.put("mail.smtp.host", host);
+        props.put("mail.smtp.port", port);
+        Session session = Session.getDefaultInstance(props);
+        session.setDebug(true);// TODO: remove before merging with master
```
Comment (drauf) [inline src/main/java/services/MailingTask.java]: .

File: src/main/java/services/MailingTask.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+        try {
+            Transport transport = session.getTransport("smtp");
+            transport.connect(host, port, from, password);
+            transport.sendMessage(msg, msg.getAllRecipients());
+            transport.close(); // TODO: Move method invocation to 'finally' block
```
Comment (danielr-kainos) [inline src/main/java/services/MailingTask.java]: .

File: src/main/java/resources/PersonResource.java [hunk 6]
```diff
@@ -53,7 +59,12 @@ public Person createPerson(Person person) {
         if (personDao.findByEmail(person.getEmail()).isEmpty()) {
             person.setActive(false);
             personDao.create(person);
-            mailService.sendEmail(person.getEmail(), person.getId());
+            try {
+                mailService.sendEmail(activationLink.createMessage(person));
+            } catch (MessagingException | IOException e) {
+                e.printStackTrace();
```
Comment (rlewan) [inline src/main/java/resources/PersonResource.java]: Please use a SLF4J Logger instead.

File: src/main/java/resources/ScheduleResource.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        this.mailService = mailService;
+    }
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response scheduleInterview(Interview interview) {
+        try {
+            mailService.sendEmail(interview.createMessage());
+        } catch (MessagingException e) {
+            e.printStackTrace();
```
Comment (rlewan) [inline src/main/java/resources/ScheduleResource.java]: Please use a SLF4J Logger instead.

File: src/main/java/services/ActivationLink.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+        Session session = Session.getDefaultInstance(new Properties());
+        MimeMessage message = new MimeMessage(session);
+
+        message.setSubject("TDP Recruitment - Activation Link");
+        message.addRecipient(Message.RecipientType.TO, new InternetAddress(person.getEmail()));
+        URL url = Resources.getResource("email/template.html");
+        String text = Resources.toString(url, Charsets.UTF_8);
+        String tempText = text.replace("{{domain}}", domain);
+        String finalText = tempText.replace("{{id}}", person.getId().toString());
```
Comment (rlewan) [inline src/main/java/services/ActivationLink.java]: You can chain `replace` calls.

File: src/main/java/services/Interview.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        UUID eventUUID = UUID.randomUUID(); //should this be random?
+        SimpleDateFormat iCalDate = new SimpleDateFormat("yyyyMMdd'T'HHmm'00Z'");
+        String now = iCalDate.format(new Date());
+        iCalDate.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        URL url = Resources.getResource("email/event_template.ics");
+        try {
+            template = Resources.toString(url, Charsets.UTF_8);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (rlewan) [inline src/main/java/services/Interview.java]: Please use a SLF4J Logger instead.

File: src/main/java/services/MailingTask.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+        session.setDebug(true);// TODO: remove before merging with master
+
+        Transport transport = null;
+
+        try {
+            transport = session.getTransport("smtp");
+            transport.connect(host, port, from, password);
+            transport.sendMessage(msg, msg.getAllRecipients());
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (rlewan) [inline src/main/java/services/MailingTask.java]: Please use a SLF4J Logger instead.

File: src/main/java/services/MailingTask.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+        } finally {
+            try {
+                if (transport != null) transport.close();
+            } catch (MessagingException e) {
+                e.printStackTrace();
```
Comment (rlewan) [inline src/main/java/services/MailingTask.java]: Please use a SLF4J Logger instead.

File: src/main/resources/app_config.yml [hunk 12]
```diff
@@ -47,4 +47,6 @@ smtpConfig:
   port: 25
   from: no-reply@kainos.com
 
-domain: http://localhost:8888
\ No newline at end of file
+domain: http://localhost:8888
+
+threadPoolSize: 10
```
Comment (rlewan) [inline src/main/resources/app_config.yml:51]: Please add a comment which explains where this variable is used.

File: src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js [hunk 13]
```diff
@@ -13,7 +14,35 @@ define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/rec
...
+        }
+
+        $scope.interviewOff = function () {
+            $scope.pairingMode = false;
+            $scope.refreshPersonsData();
+        };
+
+        $scope.createInterview = function () {
+            var data = tdprScheduleService.createInterview(slotsTimes, getSelectedPersons, $scope.params.candidate);
+            console.log(data);
```
Comment (marcint-kainos) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprWeekTableController.js]: Remove console log

File: src/main/resources/assets/js/application/recruiter/controllers/tdprCreateInterviewController.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+define(['angular', 'application/recruiter/tdprRecruiterModule'
+    , 'application/recruiter/services/tdprPersonsService'], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.controller("tdprCreateEventController", function ($scope, tdprScheduleService, tdprRecruiterSlotsService, tdprAuthService, $stateParams, $state, $q) {
+
+
+        $scope.interview = $stateParams.data;
+        $scope.interview.message = "Hi, I scheduled an interview for you.";
+        $scope.interview.organizer = angular.copy(tdprAuthService.getCurrentUser());
+        console.log($scope.interview);
```
Comment (marcint-kainos) [inline src/main/resources/assets/js/application/recruiter/controllers/tdprCreateInterviewController.js]: Remove console log.

File: src/main/java/TdpRecruitmentApplication.java [hunk 15]
```diff
@@ -3,6 +3,7 @@
 import auth.TdpRecruitmentAuthorizer;
 import auth.TdpRecruitmentPasswordStore;
 import auth.TdpRecruitmentUnauthorizedHandler;
+import com.github.dirkraft.dropwizard.fileassets.FileAssetsBundle;
```
Comment (marcint-kainos) [inline src/main/java/TdpRecruitmentApplication.java]: Remove that line

**Pull request merged**